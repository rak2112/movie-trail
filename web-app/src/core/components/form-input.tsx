import React from 'react';

export const formInput = ({ input, label, type, placeholder, meta: { touched, error, warning } }: any) => (
  <div>
    <label>{label}</label>
    <div>
      <input {...input} placeholder={placeholder || label} type={type} className={`form-control  ${(touched && error) ? 'error' : ''}`}/>
      <em>
      {touched && ((error && <span>{error}</span>) || (warning && <span>{warning}</span>))}
      </em>
      
    </div>
  </div>
);
