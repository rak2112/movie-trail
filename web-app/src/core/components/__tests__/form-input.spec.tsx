
import React from 'react';
import { create } from 'react-test-renderer';
import { FormInput } from '../form-input';

describe(`FormInput Component`, () => {
  const defaultProps = {
    input: null,
    placeholder: 'Enter your name',
    label: 'Label A',
    type: 'text',
    meta: {
      touched: false,
      error: false,
      warning: false
    }
  };

  it(`should render input without errors`, () => {
    const wrapper = create(
      <FormInput {...defaultProps}/>
    ).toJSON();
    
    expect(wrapper).toMatchSnapshot();
  });
  
  it(`should render the given input with some error`, () => {
    const props = { ...defaultProps, meta: { ...defaultProps.meta, touched: true, error: 'You have some error'}}
    const wrapper = create(
      <FormInput {...props}/>
    ).toJSON();
    
    expect(wrapper).toMatchSnapshot();
  });

  it(`should render the given input with some warning`, () => {
    const props = { ...defaultProps, meta: { ...defaultProps.meta, touched: true, warning: 'You have some warning'}}
    const wrapper = create(
      <FormInput {...props}/>
    ).toJSON();
    
    expect(wrapper).toMatchSnapshot();
  });
  
});