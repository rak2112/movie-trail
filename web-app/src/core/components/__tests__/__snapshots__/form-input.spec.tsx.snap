// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput Component should render input without errors 1`] = `
<div>
  <label>
    Label A
  </label>
  <div>
    <input
      className="form-control "
      placeholder="Enter your name"
      type="text"
    />
    <em />
  </div>
</div>
`;

exports[`FormInput Component should render the given input with some error 1`] = `
<div>
  <label>
    Label A
  </label>
  <div>
    <input
      className="form-control error"
      placeholder="Enter your name"
      type="text"
    />
    <em>
      <span>
        You have some error
      </span>
    </em>
  </div>
</div>
`;

exports[`FormInput Component should render the given input with some warning 1`] = `
<div>
  <label>
    Label A
  </label>
  <div>
    <input
      className="form-control "
      placeholder="Enter your name"
      type="text"
    />
    <em>
      <span>
        You have some warning
      </span>
    </em>
  </div>
</div>
`;
